ml/Host.cc

```diff
@@ -0,0 +1,297 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "Config.h"
+#include "Host.h"
+
+#include "json/single_include/nlohmann/json.hpp"
+
+using namespace ml;
+
+static void updateDimensionsChart(RRDHOST *RH,
+                                  collected_number NumNormalDimensions,
+                                  collected_number NumAnomalousDimensions) {
+    static thread_local RRDSET *RS = nullptr;
+    static thread_local RRDDIM *NumNormalDimensionsRD = nullptr;
+    static thread_local RRDDIM *NumAnomalousDimensionsRD = nullptr;
+
+    if (!RS) {
+        RS = rrdset_create(
+            RH, // host
+            "anomaly_detection", // type
+            "dimensions", // id
+            NULL, // name
+            "anomaly_detection", // family
+            NULL, // ctx
+            "Anomaly detection dimensions", // title
+            "dimensions", // units
+            "netdata", // plugin
+            "ml", // module
+            39183, // priority
+            Cfg.UpdateEvery, // update_every
+            RRDSET_TYPE_LINE // chart_type
+        );
+
+        NumNormalDimensionsRD = rrddim_add(RS, "normal", NULL,
+                1, 1, RRD_ALGORITHM_ABSOLUTE);
+        NumAnomalousDimensionsRD = rrddim_add(RS, "anomalous", NULL,
+                1, 1, RRD_ALGORITHM_ABSOLUTE);
+    } else
+        rrdset_next(RS);
+
+    rrddim_set_by_pointer(RS, NumNormalDimensionsRD, NumNormalDimensions);
+    rrddim_set_by_pointer(RS, NumAnomalousDimensionsRD, NumAnomalousDimensions);
+
+    rrdset_done(RS);
+}
+
+static void updateRateChart(RRDHOST *RH, collected_number AnomalyRate) {
+    static thread_local RRDSET *RS = nullptr;
+    static thread_local RRDDIM *AnomalyRateRD = nullptr;
+
+    if (!RS) {
+        RS = rrdset_create(
+            RH, // host
+            "anomaly_detection", // type
+            "anomaly_rate", // id
+            NULL, // name
+            "anomaly_detection", // family
+            NULL, // ctx
+            "Percentage of anomalous dimensions", // title
+            "percentage", // units
+            "netdata", // plugin
+            "ml", // module
+            39184, // priority
+            Cfg.UpdateEvery, // update_every
+            RRDSET_TYPE_LINE // chart_type
+        );
```

Comment:
@ilyam8 can you help me understand what I need to do in order to show the information that @andrewm4894 wants here: [https://github.com/vkalintiris/netdata/pull/8/files](https://github.com/vkalintiris/netdata/pull/8/files)?

Applying @andrewm4894 's patch does not add the information to the UI.

Reply:
Hey, there is  ["Customize the standard dashboard"](https://learn.netdata.cloud/docs/dashboard/customize#customize-the-standard-dashboard) doc.

TL;DR

 - [netdataDashboard.menu](https://github.com/netdata/netdata/blob/07d3588871f7fd8f4b05a0ea70481181e413e4e2/web/gui/dashboard_info.js#L13) - for top level sections, it matches **type** part of the chart `type.id`.
 - [netdataDashboard.submenu](https://github.com/netdata/netdata/blob/07d3588871f7fd8f4b05a0ea70481181e413e4e2/web/gui/dashboard_info.js#L673) - for submenus, it matches `family` of the chart.
 - [netdataDashboard.context](https://github.com/netdata/netdata/blob/07d3588871f7fd8f4b05a0ea70481181e413e4e2/web/gui/dashboard_info.js#L957) - for charts, it matches `context` of the chart.

Use the appropriate variable depends on your need.

Reply:
Keep in mind that you need to "Hard reload" the page after updating the `dashboard_info.js` file in order to see your changes.

Reply:
Ok, I didn't notice there is a `_` in the **type**, that is the problem.

Reply:
Thanks! That was it. However, after removing `_` from the type, I'm still not getting the list of individual charts in the submenu:

![Screenshot at 2021-09-28 14-32-05](https://user-images.githubusercontent.com/25480816/135078954-66494bf5-f382-4dce-bb5f-3b90bb6b80a6.png)

Normally, I would expect the following list to get displayed under the "Anomaly Detection" menu:

- `dimensions`
- `anomaly_rate`
- `detector_window`
- `detector_events`
- `prediction_stats`
- `training_stats`

Any suggestion on how to fix this?

Reply:
You have changed the chart families to `anomalydetection`, that is why there is only one _submenu_ (submenu == family).

Reply:
+1 Thanks @ilyam8